### PR TITLE
Add simple cache prefix decorator

### PR DESCRIPTION
**Pull request by ndobromirov**
| Question      | Answer
| ------------- | ------
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? |no
| Tests pass?   | yes
| Fixed tickets | 111
| License       | MIT
| Doc PR        | reference to the documentation PR, if any

### Description
Added simple-cache PSR package as a dependency.
Added the new decorator for simple cache compliant adapters.

### TODO
* [x] Add tests
* [x] Updated Changelog.md

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 /vendor/
 composer.lock
 phpunit.xml
+nbproject
```
Comment (prisis) [inline .gitignore]: you should remove this

File: src/Prefixed/PrefixedSimpleCache.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        $this->cache  = $simpleCache;
+        $this->prefix = $prefix;
+    }
+
+    /**
+     * Add namespace prefix on the key.
+     *
+     * @param array $keys
+     */
+    private function prefixValue(&$key)
```
Comment (prisis) [inline src/Prefixed/PrefixedSimpleCache.php]: Move this to a trait :) the other private too

File: src/Prefixed/PrefixedSimpleCache.php [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+use Psr\SimpleCache\CacheInterface;
+
+/**
+ * PrefixedSimpleCache.
+ *
+ * Prefixes all cache keys with a string.
+ *
+ * @author ndobromirov
+ */
+class PrefixedSimpleCache implements CacheInterface
```
Comment (prisis) [inline src/Prefixed/PrefixedSimpleCache.php]: Add a test for this class :)

**Comment by ndobromirov** [comment]
Rough idea - the get multiple returns key => values list.
When requesting we will have to add the prefix, when returning it should likely strip it.

Otherwise get -> set -> get -> set will aggregate the prefix...

**Comment by Nyholm** [comment]
Hey. How is it going? Im about to release 1.0.0 of the prefixed cache. Do you want this feature to be included or do you want to have it in 1.1.0?

**Comment by ndobromirov** [comment]
Do not block 1.0.0 with this. I am pretty sure that it's something small that needs to be tried, but I have problems with local set-up on an old Ubuntu being on PHP 5.5...

Is there a release calendar, or once this is done you can release 1.1.0?

**Comment by Nyholm** [comment]
Im sorry about PHP5.5. :(

> Is there a release calendar, or once this is done you can release 1.1.0?

No, we release when ever we want. And of course we should try to release your fix ASAP.

**Comment by ndobromirov** [comment]
This just attempts to fix the tests passing. Once that's OK, I will re-base from the latest master.

**Comment by ndobromirov** [comment]
What is Travis doing that long?

**Comment by prisis** [comment]
It should work now :)

**Comment by ndobromirov** [comment]
Small comment change.
Rebased the branch from https://github.com/php-cache/cache/tree/master.

**Comment by ndobromirov** [comment]
I think the Travis on one of the machines died. It did not reach to execute tests at all.
The rest pass :). 

If t here is more to do on this, let me know :).

**Comment by prisis** [comment]
rebuild is started :dancer:

**Comment by ndobromirov** [comment]
This failed on a command that succeeded last time. There is some kind of Travis issue.

**Comment by prisis** [comment]
all green :)

now you need to update the Changelog.md and @Nyholm or i can merge it

**Pull request merged**